src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -483,6 +514,19 @@ class DocBaseViewer extends BaseViewer {
         return true;
     }
 
+    /**
+     * Sets the start page the preview should start at
+     *
+     * @param {number} startPageNum - the start page number to start the preview at
+     * @param {number} pagesCount - the number of pages in the document
+     * @return {void}
+     */
+    setStartPage(startPageNum, pagesCount) {
+        if (startPageNum && startPageNum > 0 && startPageNum <= pagesCount) {
```

Comment:
Would it be safer to `parseInt(startPageNum)` and check to make sure it isn't `NaN`? `true` can cause this logic to run, as `startPageNum > 0 && startPageNum <= pagesCount` equates to `true`, where `startPageNum = true`. Also, as a bonus, we'll have floats and strings parsed to integers and avoid passing strange data to `this.cachePage()`

Reply:
Also, I don't think you need to have any of this logic at all. `setPage()` internally manages valid pages. If you have a value for page, it's safe to blindly pass that to `setPage()`